src/python/foglamp/translators/omf_translator.py

```diff
@@ -0,0 +1,732 @@
+
+        #
+        # Useful for an execution
+        #
+        SELECT MAX(ID) FROM foglamp.readings WHERE id >= 93491;
+
+        UPDATE foglamp.streams SET reading_id=106928, ts=now() WHERE id=1;
+
+        SELECT * FROM foglamp.streams;
+
+        SELECT * FROM foglamp.readings WHERE id > 98021 ORDER by USER_ts;
+
+        SELECT * FROM foglamp.readings WHERE id >= 98021 and reading ? 'lux' ORDER by USER_ts;
+
+
+Todo:
+    - it should evolve using the DB layer
+    - only part of the code is using async
+
+"""
+
+# Import packages
+import json
+import time
+import requests
+import sys
+import datetime
+
+import logging
+import logging.handlers
+
+# Import packages - DB operations
+import psycopg2
+import asyncio
+import aiopg
+import aiopg.sa
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+
+# Module information
+__module__    = "OMF Translator"
+__prg_text__  = ", for Linux (x86_64)"
+__company__   = "2017 DB SOFTWARE INC."
+
+__author__    = "2017 DB SOFTWARE INC."
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__   = "Apache 2.0"
+__version__   = "1.0.17"
+
+DB_URL     = 'postgresql://foglamp:foglamp@localhost:5432/foglamp'
+"""DB references"""
+
+LOG_SCREEN = True
+"""Enable/Disable screen messages"""
+
+
+_message_list  = {
+    # Information messages
+    "i000001": "operation successfully completed",
+    "i000002": __module__ + " - Started",
+    "i000003": __module__ + " - Execution completed.",
+
+    # Warning / Error messages
+    "e000001": __module__ + " - generic error.",
+    "e000002": __module__ + " - cannot retrieve the starting point for sending operation - error details |{0}|.",
+    "e000003": __module__ + " - cannot update the reached position - error details |{0}|.",
+    "e000004": __module__ + " - cannot complete the sending operation - error details |{0}|.",
+    "e000005": __module__ + " - cannot configure the logging mechanism. - error details |{0}|.",
+    "e000006": __module__ + " - cannot initialize the plugin. - error details |{0}|.",
+    "e000007": __module__ + " - an error occurred during the OMF request. - error details |{0}|.",
+    "e000008": __module__ + " - an error occurred during the OMF's objects creation. - error details |{0}|.",
+    "e000009": __module__ + " - cannot retrieve information about the sensor.",
+    "e000010": __module__ + " - unable ro create the JSON message.",
+
+}
+"""Used messages"""
+
+# Logger
+_log       = ""
+_handler   = ""
+_formatter = ""
+
+# SA related
+_readings_values_tbl = sa.Table(
+    'readings',
+    sa.MetaData(),
+    sa.Column('id', sa.BigInteger, primary_key=True),
+    sa.Column('asset_code', sa.types.VARCHAR(50)),
+    sa.Column('read_key', sa.types.VARCHAR(50)),
+    sa.Column('user_ts', sa.types.TIMESTAMP),
+    sa.Column('reading', JSONB))
+"""Contains the information to send to OMF"""
+
+
+# PI Server references
```

Comment:
add a link in comment for reference I think http://omf-docs.readthedocs.io/en/v1.0/Data_Msg_Sample.html#data-example